### PR TITLE
Feature/extend manager to use channel slug

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds logic to plugins manager responsible for distributing the hook calls for plugin assigned to the channels

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/plugins/mutations.py [hunk 1]
```diff
@@ -45,6 +49,7 @@ class Meta:
     @classmethod
     def perform_mutation(cls, root, info, **data):
         plugin_id = data.get("id")
+        channel_slug = data.get("channel")
         data = data.get("input")
         manager = info.context.plugins
         plugin = manager.get_plugin(plugin_id)
```
Comment (IKarbowiak) [inline saleor/graphql/plugins/mutations.py:55]: Shouldn't we have here the same channel validation as it was added to `PaymentInitialize`?

Comment (korycins) [reply saleor/graphql/plugins/mutations.py:55]: No. We still have some plugins which are configurable globally (like Emails, Webhooks). PaymentPlugins by default are configurable by channel. 
So here, it is fine if I will receive the channel as a None value

Comment (patrys) [reply saleor/graphql/plugins/mutations.py:55]: Slightly off-topic, but wouldn't it make sense to configure transactional emails on a per-channel basis?

File: saleor/graphql/shop/types.py [hunk 2]
```diff
@@ -207,8 +216,12 @@ class Meta:
         )
 
     @staticmethod
-    def resolve_available_payment_gateways(_, info, currency: Optional[str] = None):
-        return info.context.plugins.list_payment_gateways(currency=currency)
+    def resolve_available_payment_gateways(
+        _, info, currency: Optional[str] = None, channel: Optional[str] = None
+    ):
+        return info.context.plugins.list_payment_gateways(
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/graphql/shop/types.py:222]: Probably I will drop currencies in separate PR

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/plugins/mutations.py (hunk 1 above)
Comment (korycins) [reply saleor/graphql/plugins/mutations.py:55]: We can convert them to per channel but I would prefer to do that in separate PR

**Pull request merged**